### DISCUSSION TITLE
[Question][Hw2] How do we know that the previous state is enough?

### DISCUSSION THREAD

**Discussion by qawbecrdtey** [prosyslab-classroom/cs524-program-analysis#372] [Q&A]
There might be an issue if we don't store every states before the current, but the current implementation seems to only store the state of the immediately preceded block. How do we know that is sufficient?

**Comment by KAIST-JongchanPark** [comment]
I'm very sorry, but I cannot fully understand what your "issue" is...

Could you explain your question one more time in detail?

It would be helpful if there is any example.

**Comment by qawbecrdtey** [comment]
`prev` only holds the state from the preceding block.
In some complex program, there might be a case where the single previous state is not enough, but it seems like the program runs without problem.
Why is this the case? Why is the preceded state sufficient for the design with the phi node?

**Comment by goodtaeeun** [comment]
I guess what you mean by the "case where the single previous state is not enough" is a case where there are multiple incoming locations for a single instruction. In this case, all the information of the incoming blocks is well handled with the phi node.

**Comment by qawbecrdtey** [comment]
Thanks for the answer. Maybe I should think about it more.
